lib/websocket.js

```diff
@@ -181,13 +216,98 @@ function init (env, ctx, server) {
           data.data.created_at = new Date().toISOString();
         }
         
-        ctx.store.db.collection(collection).insert(data.data, function insertResult(err, doc) {
-          if (callback) {
-            callback(doc.ops);
+        // treatments deduping
+        if (data.collection === 'treatments') {
+          var query  = {
+            created_at: data.data.created_at
+            , eventType: data.data.eventType
+          };
+         
+         // try to find exact match
+          ctx.store.db.collection(collection).find(query).toArray(function findResult(err, array) {
+            if (err || array.length > 0) {
+                console.log(LOG_DEDUP + 'Exact match');
+                if (callback) {
+                  callback([array[0]]);
+                }
+                return;
+            }
+                
+            var query_similiar  = {
+              created_at: {$gte: new Date(new Date(data.data.created_at).getTime() - maxtimediff).toISOString(), $lte: new Date(new Date(data.data.created_at).getTime() + maxtimediff).toISOString()}
+              , eventType: data.data.eventType
+            };
+            if (data.data.insulin) {
+              query_similiar.insulin = data.data.insulin;
+            }
+            if (data.data.carbs) {
+              query_similiar.carbs = data.data.carbs;
+            }
+            if (data.data.happ_id) {
+              query_similiar.happ_id = data.data.happ_id;
+            }
+console.log('Looking for similiar', query);
+            // try to find similiar
+            ctx.store.db.collection(collection).find(query_similiar).toArray(function findResult(err, array) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Function 'findResult' is already declared in line 227.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1885581608/issues/source?bid=2452379&fileBranchId=3084282#l251)